core/trino-main/src/main/java/io/trino/operator/DriverFactory.java

```diff
@@ -38,7 +38,7 @@
     private final OptionalInt driverInstances;
 
     @GuardedBy("this")
-    private boolean noMoreDrivers;
+    private volatile boolean noMoreDrivers; // must synchronize between createDriver() and noMoreDrivers(), but isNoMoreDrivers() need not synchronize
```

Comment:
This change seems unrelated and should be moved to a separate commit

Reply:
Will do.